lib/galaxy/files/sources/__init__.py

```diff
@@ -53,7 +53,12 @@ class FilesSourceProperties(TypedDict):
 
 
 class FilesSourceOptions:
-    """Options to control behaviour of filesource operations, such as realize_to and write_from"""
+    """Options to control behavior of file source operations, such as realize_to, write_from and list."""
+
+    # Indicates access to the FS operation with intent to write.
+    # A file source can be "writeable" but, for example, some directories (or elements) may be restricted or read-only
```

Comment:
I see, in that case this seems like something you'd apply when listing the contents ... I'd still call it writable though ?

Reply:
Yeah, I wanted to differentiate it from the general `writeable` flag for the whole plugin. This `write_intent` makes only sense when the plugin is already "writeable" and provides just a way of signaling that "we are currently browsing the remote source to write a file on it, so please return only those directories that can be written to" as opposed from when we are browsing to import a file, which should list all o them, including read-only directories (or records).

Happy to change it to something more meaningful, I just think "writeable" could be confused with the general flag. Any other naming idea?

Reply:
I don't think this has to be confusing, one is for the plugin, one is for the listing. They're not used in the same call, or are they ?

Reply:
You are right, that is a good point. And besides, the API docs will clarify it anyway, I'll change it to `writeable` :+1: